File: DNA.py
# Rosalind ID: DNA
# Title: Counting DNA Nucleotides

# Open the data file
with open('rosalind_dna.txt') as dna_file:
	dna = dna_file.read()

# Count the number of each nucleotide
counts = []
for x in ['A', 'C', 'G', 'T']:
	counts.append(str(dna.count(x)))

# Print counts
print(' '.join(counts))

File: DNA.R
# Rosalind ID: DNA
# Title: Counting DNA Nucleotides

# Open dna file
dna <- readChar("~/Rosalind_Python/rosalind_dna.txt", nchars = file.info("~/Rosalind_Python/rosalind_dna.txt")$size)

# Check structure of the file dna
str(dna)

# Load library stringr
library(stringr)

# Count A, C, G, T in dna
str_count(dna, c("A","C","G","T"))
